src/test/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshServiceTest.java

```diff
@@ -0,0 +1,265 @@
+            scanDelayInSeconds,
+            scanIntervalInMilliseconds,
+            pollingCycleMaximumDurationInSeconds);
+    }
+
+    @Test
+    void when_IntervalPassedAndMessagesFound_Then_DownloadAndPublishAllMessages() {
+        when(meshMailBoxScheduler.hasTimePassed(scanDelayInSeconds)).thenReturn(true);
+        when(meshMailBoxScheduler.isEnabled()).thenReturn(true);
+        when(meshClient.getInboxMessageIds()).thenReturn(List.of(MESSAGE_ID1, MESSAGE_ID2));
+        when(meshClient.getEdifactMessage(MESSAGE_ID1)).thenReturn(meshMessage1);
+        when(meshClient.getEdifactMessage(MESSAGE_ID2)).thenReturn(meshMessage2);
+
+        meshService.scanMeshInboxForMessages();
+
+        verify(meshClient).authenticate();
+        verify(meshClient).getEdifactMessage(MESSAGE_ID1);
+//        verify(inboundQueueService).publish(meshMessage1);
+        verify(meshClient).acknowledgeMessage(MESSAGE_ID1);
+
+        verify(meshClient).getEdifactMessage(MESSAGE_ID2);
+//        verify(inboundQueueService).publish(meshMessage1);
+        verify(meshClient).acknowledgeMessage(MESSAGE_ID2);
+
+        verify(conversationIdService, times(2)).applyRandomConversationId();
+        verify(conversationIdService, times(2)).resetConversationId();
+    }
+
+    @Test
+    void when_IntervalHasPassed_DurationExceeded_Then_StopDownloading() {
+        when(meshMailBoxScheduler.hasTimePassed(scanDelayInSeconds)).thenReturn(true);
+        when(meshMailBoxScheduler.isEnabled()).thenReturn(true);
+        when(meshClient.getInboxMessageIds()).thenReturn(List.of(MESSAGE_ID1, MESSAGE_ID2));
+        when(meshClient.getEdifactMessage(any())).thenAnswer((invocation) -> {
+            Thread.sleep((pollingCycleMaximumDurationInSeconds + 1) * 1000L); // ensure first download exceeds duration
+            return meshMessage1;
+        });
+
+        meshService.scanMeshInboxForMessages();
+
+        verify(meshClient).authenticate();
+        verify(meshClient).getEdifactMessage(MESSAGE_ID1);
+//        verify(inboundQueueService).publish(meshMessage1);
+        verify(meshClient).acknowledgeMessage(MESSAGE_ID1);
+//        verifyNoMoreInteractions(meshClient, inboundQueueService); // second message not downloaded and published
+
+        verify(conversationIdService).applyRandomConversationId();
+        verify(conversationIdService).resetConversationId();
+    }
+
+    @Test
+    void when_IntervalPassedAndRequestToGetMessageListFails_Then_DoNotPublishAndAcknowledgeMessages() {
+        when(meshMailBoxScheduler.hasTimePassed(scanDelayInSeconds)).thenReturn(true);
+        when(meshMailBoxScheduler.isEnabled()).thenReturn(true);
+        when(meshClient.getInboxMessageIds()).thenThrow(new MeshApiConnectionException("error"));
+
+        assertThatThrownBy(meshService::scanMeshInboxForMessages).isExactlyInstanceOf(MeshApiConnectionException.class);
+
+        verify(meshClient).authenticate();
+        verify(meshClient, times(0)).getEdifactMessage(any());
```

Comment:
There's a `never()` we can use instead of `times(0)` to be a bit more natural languagey.